runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2238,14 +2238,31 @@ J9::Z::TreeEvaluator::inlineMathFma(TR::Node *node, TR::CodeGenerator *cg)
    TR_ASSERT_FATAL(node->getNumChildren() == 3,
    "In function inlineMathFma, the node at address %p should have exactly 3 children, but got %u instead", node, node->getNumChildren());
 
-   TR::Register      * targetRegister      = cg->allocateRegister(TR_FPR);
-
    TR::Register      * v1      = cg->evaluate(node->getFirstChild());
    TR::Register      * v2      = cg->evaluate(node->getSecondChild());
-   TR::Register      * v3      = cg->evaluate(node->getThirdChild());
+   TR::Register      * v3;
+   TR::Register      * targetRegister;
 
-   uint8_t mask6 = getVectorElementSizeMask(TR::DataType::getSize(node->getDataType()));
-   generateVRReInstruction(cg, TR::InstOpCode::VFMA, node, targetRegister, v1, v2, v3, mask6, 0);
+   if (cg->getSupportsVectorRegisters() && (node->getDataType() == TR::Double || node->getDataType() == TR::Float && cg->comp()->target().cpu.getSupportsVectorFacilityEnhancement1()))
+      {
+      v3 = cg->evaluate(node->getThirdChild());
```

Comment:
Only the Vector version of the instruction needs 4 registers. We can reduce the scope of the `v3` here to the `if` block where we are using `VFMA`.